src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -1293,63 +1294,80 @@ learnPointsTo ::
-               --                          ])
-               -- PP.<$$> PP.text (show err)
+            case res of
+              Crucible.NoErr pred_ res_val -> do
+                pred_' <- case maybe_cond of
+                  Just cond -> do
+                    cond' <- instantiateExtResolveSAWPred sc cc (ttTerm cond)
+                    liftIO $ W4.impliesPred sym cond' pred_
+                  Nothing -> return pred_
+                addAssert pred_' $ Crucible.SimError loc $
+                  Crucible.AssertFailureSimError (show $ PP.vcat $ summarizeBadLoad) ""
+                pure Nothing <* matchArg opts sc cc spec prepost res_val memTy val'
+              _ -> do
+                -- When we have a concrete failure, we do a little more computation to
+                -- try and find out why.
+                let (blk, _offset) = Crucible.llvmPointerView ptr1
+                pure $ Just $ PP.vcat . (summarizeBadLoad ++) $
+                  case W4.asNat blk of
+                    Nothing -> [PP.text "<Read from unknown allocation>"]
+                    Just blk' ->
+                      let possibleAllocs =
+                            Crucible.possibleAllocs blk' (Crucible.memImplHeap mem)
+                      in [ PP.text $ unwords
+                           [ "Found"
+                           , show (length possibleAllocs)
+                           , "possibly matching allocation(s):"
+                           ]
+                         , bullets '-' (map Crucible.ppSomeAlloc possibleAllocs)
+                         ]
+                      -- This information tends to be overwhelming, but might be useful?
+                      -- We should brainstorm about better ways of presenting it.
+                      -- PP.<$$> PP.text (unwords [ "Here are the details on why reading"
+                      --                          , "from each matching write failed"
+                      --                          ])
+                      -- PP.<$$> PP.text (show err)
+       SymbolicSizeValue expected_arr_tm expected_sz_tm ->
+         do maybe_allocation_array <- liftIO $
+              Crucible.asMemAllocationArrayStore sym Crucible.PtrWidth ptr1 (Crucible.memImplHeap mem)
+            liftIO $ putStrLn $ "***" ++ show maybe_allocation_array
+            case maybe_allocation_array of
+              Just (arr, sz)
+                | Crucible.LLVMPointer _ off <- ptr1
+                , Just 0 <- BV.asUnsigned <$> W4.asBV off ->
+                do arr_tm <- liftIO $ Crucible.toSC sym arr
+                   instantiateExtMatchTerm sc cc (spec ^. MS.csLoc) prepost arr_tm (ttTerm expected_arr_tm)
+                   sz_tm <- liftIO $ Crucible.toSC sym sz
+                   instantiateExtMatchTerm sc cc (spec ^. MS.csLoc) prepost sz_tm (ttTerm expected_sz_tm)
+                   return Nothing
+              _ -> return $ Just $ PP.text ""
```

Comment:
This is an error case, right? If so, it should have a meaningful message.

Reply:
done